asciidoc-extensions/src/main/resources/ExampleNumberingProcessor.rb

```diff
@@ -0,0 +1,43 @@
+# encoding: UTF-8
+require 'asciidoctor'
+require 'asciidoctor/extensions'
+
+class ExampleIndexes
+  attr_accessor :sectionIndex
+  @sectionIndex
+  attr_accessor :exampleIndex
+  @exampleIndex
+  attr_accessor :tableIndex
+  @tableIndex
+  def initialize(sect)
+    @sectionIndex = sect
+    @tableIndex = 1
+    @exampleIndex = 1
+  end
+end
+
+class ExampleNumberingProcessor < Asciidoctor::Extensions::Treeprocessor
```

Comment:
I did started with Java at first and didn't want to write it in Ruby :)
But there were two problems with writing it in Java:
- first one is that asciidoctorj classes API do not provide access to the needed fields. I've solved this with using some reflection. But I didn't found a nice way to retrieve a needed parameter, as all the attributes were represented by `RubyArray` (or something like that). And after I more or less solved the problem with finding the needed attribute - the second problem occurred
- second one is that when I've added other extensions (`CustomRoleBlockProcessor` in particular) they replaced document nodes with some sort of anonymous interface implementations (at least I think so :) ) and those blocks that were `Section` before were not sections anymore. As a result they were ignored by this numbering extension and resulting documentation had complete sections with wrong numbering style.

Thinking more about the second problem I think it'll be possible to rewrite the java code that I had to use node_name or style attributes to filter for sections rather than using `instanceof`.

I agree that having it in Java would be much better (especially when knowing that I'm not a Ruby dev :) and never used Ruby for coding till now :) ), but on the other hand I do see a couple of benefits having it written in Ruby:
- it's really much more cleaner implemented (without reflection and guessing which attribute should be modified)
- it's more like a native extension rather than going through additional proxy 

With that said - if you'd like I can give it one more try to implement it in Java :)